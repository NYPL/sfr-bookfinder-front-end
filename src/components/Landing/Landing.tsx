import React from "react";
import {
  Breadcrumbs,
  BreadcrumbsTypes,
  Heading,
  HeadingLevels,
  Hero,
  HeroTypes,
  List,
  ListTypes,
  Template,
  TemplateBreakout,
  TemplateContent,
  TemplateContentPrimary,
  TemplateContentTop,
} from "@nypl/design-system-react-components";
import SearchForm from "~/src/components/SearchForm/SearchForm";
import Subjects from "~/config/subjectListConfig";
import Link from "~/src/components/Link/Link";
import { defaultBreadcrumbs } from "~/src/constants/labels";

const LandingPage: React.FC<any> = () => {
  const subHeader = (
    <div className="hero__body-text">
      Find millions of digital books for research from multiple sources
      world-wide--all free to read, download, and keep. No library card
      required. This is an early beta test, so we want your feedback!{" "}
      <Link to="/about" color="ui.white">
        Read more about the project
      </Link>
      .
    </div>
  );
  return (
    <Template>
      <TemplateBreakout>
        <Breadcrumbs
          breadcrumbsType={BreadcrumbsTypes.Research}
          breadcrumbsData={defaultBreadcrumbs}
        />
        <div
          aria-label="Digital Research Books Beta"
          className="hero-container"
        >
          <Hero
            backgroundColor="section.research.primary"
            heroType={HeroTypes.Tertiary}
            heading={
              <Heading level={HeadingLevels.One} id="tertiary-hero">
                <>
                  Digital Research Books <sup>Beta</sup>
                </>
              </Heading>
            }
            subHeaderText={subHeader}
          />
        </div>
      </TemplateBreakout>
      <TemplateContent>
        <TemplateContentTop>
          Search the World's Research Collections
          <SearchForm />
        </TemplateContentTop>
        <TemplateContentPrimary>
          <Heading level={HeadingLevels.Two}>Search Examples</Heading>

          <List type={ListTypes.Unordered} id="subject-list">
            {Subjects.map((sub: any) => (
              <li key={`subject-link-${sub.url}`}>
                <Link to={sub.url}>{sub.text}</Link>
              </li>
            ))}
          </List>
        </TemplateContentPrimary>
      </TemplateContent>
    </Template>
  );
};

export default LandingPage;
