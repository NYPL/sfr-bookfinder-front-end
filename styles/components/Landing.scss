.hero__body-text {
    @include space-inset-m;
}
.searchbar-container {
    @include breakout;
    background-color: var(--ui-gray-light);
}

.searchbar { 
    @include wrapper;
    @include space-inset-m;

    display: flex;
    flex-direction: column;

    @media (max-width: $breakpoint-small) {
        flex-direction: row;
    }
}
.bar-content {
    flex: 1 0 auto;
}
.advanced-search-message {
    flex: 0 1 auto;
    align-self: flex-end;
}

.search-examples {
    @include space-inset-m;
    & li {
       @include space-stack-s; 
    }
}

