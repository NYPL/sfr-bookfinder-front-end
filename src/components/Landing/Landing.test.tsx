import React from "react";
import Landing from "./Landing";
import { shallow } from "enzyme";

test("renders Landing page correctly", () => {
  const about = shallow(<Landing />);
  expect(about).toMatchSnapshot();
});
