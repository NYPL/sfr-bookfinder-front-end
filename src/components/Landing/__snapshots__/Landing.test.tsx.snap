// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Landing page correctly 1`] = `
<main
  className="main"
>
  <div
    className="content-header"
  >
    <Breadcrumbs
      breadcrumbs={
        Array [
          Object {
            "text": "Digital Research Books Beta",
            "url": "/",
          },
        ]
      }
    />
    <div
      aria-label="Digital Research Books Beta"
      className="main-promo"
    >
      <Hero
        heading={
          <Heading
            blockName="hero"
            id="1"
            level={1}
            text="Digital Research Books Beta"
          />
        }
        heroType="SECONDARY"
        subHeaderText={
          <div
            className="hero__body-text"
          >
            Find millions of digital books for research from multiple sources world-wide--all free to read, download, and keep. No library card required. This is an early beta test, so we want your feedback!
             
            <Link>
              <ForwardRef
                to="/about"
              >
                <React.Fragment>
                  Read more about the project
                </React.Fragment>
              </ForwardRef>
            </Link>
            .
          </div>
        }
      />
    </div>
  </div>
  <div
    className="content-primary"
  >
    <SearchForm />
  </div>
  <div
    className="content-primary search-examples"
  >
    <Heading
      level={2}
    >
      Search Examples
    </Heading>
    <List
      id="subject-list"
      modifiers={
        Array [
          "no-list-styling",
        ]
      }
      type="ul"
    >
      <li
        key="subject-link-/search?queries=[{\\"query\\"%3A\\"Sub-Saharan Africa\\"%2C\\"field\\"%3A\\"subject\\"}]"
      >
        <Link>
          <ForwardRef
            to="/search?queries=[{\\"query\\"%3A\\"Sub-Saharan Africa\\"%2C\\"field\\"%3A\\"subject\\"}]"
          >
            Subject: Sub-saharan Africa
          </ForwardRef>
        </Link>
      </li>
      <li
        key="subject-link-/search?showQueries=[{\\"query\\"%3A\\"Climate Change\\"%2C\\"field\\"%3A\\"keyword\\"}]"
      >
        <Link>
          <ForwardRef
            to="/search?showQueries=[{\\"query\\"%3A\\"Climate Change\\"%2C\\"field\\"%3A\\"keyword\\"}]"
          >
            Keyword: climate change
          </ForwardRef>
        </Link>
      </li>
      <li
        key="subject-link-/search?showQueries=[{\\"query\\"%3A\\"New York City\\"%2C\\"field\\"%3A\\"title\\"}]"
      >
        <Link>
          <ForwardRef
            to="/search?showQueries=[{\\"query\\"%3A\\"New York City\\"%2C\\"field\\"%3A\\"title\\"}]"
          >
            Title: New York City
          </ForwardRef>
        </Link>
      </li>
      <li
        key="subject-link-/search?queries=[{\\"query\\"%3A\\"Corellli%2C Marie\\"%2C\\"field\\"%3A\\"author\\"}]"
      >
        <Link>
          <ForwardRef
            to="/search?queries=[{\\"query\\"%3A\\"Corellli%2C Marie\\"%2C\\"field\\"%3A\\"author\\"}]"
          >
            Author: Corelli, Marie
          </ForwardRef>
        </Link>
      </li>
      <li
        key="subject-link-/search?queries=[{\\"query\\"%3A\\"cookbooks\\"%2C\\"field\\"%3A\\"keyword\\"}]"
      >
        <Link>
          <ForwardRef
            to="/search?queries=[{\\"query\\"%3A\\"cookbooks\\"%2C\\"field\\"%3A\\"keyword\\"}]"
          >
            Keyword: cookbooks
          </ForwardRef>
        </Link>
      </li>
    </List>
  </div>
</main>
`;
